test/helpers/framework.go

```diff
@@ -260,19 +260,25 @@ func (tc *E2ETestFramework) waitForElasticsearchPods(retryInterval, timeout time
 			clolog.Error(err, "Error listing elasticsearch pods")
 			return false, nil
 		}
-		if len(pods.Items) == 0 {
+		numberOfPods := len(pods.Items)
```

Comment:
This is probably not the expected number of pods. We deploy ES cluster by creating an instance of Elasticsearch CR. better to compare with the length of `elasticsearch.spec.nodes.nodeCount`

but i think in the e2e test we create only one ES

Reply:
Hi @vimalk78 thanks for this comment ... agree ... but ... I see that this is a fixed/ hardcoded value of 1 

https://github.com/openshift/cluster-logging-operator/blob/master/test/helpers/elasticsearch.go#L194

The CR is not saved to any variable, so to know nodeCount there is a need to read it back from k8s (by CR name?). 

https://github.com/openshift/cluster-logging-operator/blob/master/test/helpers/elasticsearch.go#L256

Taking into account the rest of the waiting functions (look on fluentd that do not check the count as expected) this enhancement can be postponed to following PR

Reply:
@vimalk78 I see that this should theoretically be the value in tc.ClusterLogging.Spec.LogStore.NodeCount but as said above in some cases this is saved and in some other it is not ... so can't count on that
		
`